### DISCUSSION TITLE
Loading Form Data on All PDF Pages with an Event

### DISCUSSION THREAD

**Discussion by pramane2309** [stephanrauh/ngx-extended-pdf-viewer#2354] [Q&A]
Hello,

I would like to know if it is possible with an event to load all the formData onto the PDF. Currently, when I call annotationLayerRendered, it loads the data onto pages 1 and 2 of the PDF, but not on page 3 and beyond. You have to scroll to these pages to populate them. Is there another solution, please?

**Comment by stephanrauh** [comment]
Basically, you've got a problem with lazy loading. That's a feature that's deeply ingrained in the DNA of pdf.js, so it's hard to circumvent.

However, you can enforce pre-rendering. That's still asynchronous, so I guess it's still cumbersome for you, but maybe it helps. Have a look at https://pdfviewer.net/extended-pdf-viewer/prerendering and open the "TypeScript" tab. It shows how to talk pdf.js into rendering more pages in advance than it does by default. You can also use the idea to render the entire document. That includes the annotation layer, so I hope it solves your issue.

However, if you render too many pages, pdf.js starts evicting pages from the cache. But I don't think that happens with real-life forms.
